File: 2022.convert-1d-array-into-2d-array/solution.go
func construct2DArray(original []int, m int, n int) [][]int {
    if len(original) != m*n {
        return nil
    }
    res := make([][]int, m)
    for i := range m {
        res[i] = original[i*n : (i+1) * n]
    }
    return res
}

File: 2022.convert-1d-array-into-2d-array/solution.kt
class Solution {
    fun construct2DArray(original: IntArray, m: Int, n: Int): Array<IntArray> {
        return if (original.size != m * n) emptyArray()
        else Array(m) { original.sliceArray(it * n until (it + 1) * n) }
    }
}